apache2/persist_dbm.c

```diff
@@ -599,7 +598,7 @@ static int collection_store_ex(int db_option, modsec_rec *msr, apr_table_t *col)
         if(ag_dbm == NULL) {
             //Create the DB
             root_dcfg = msr->dcfg1->root_config;
-            sprintf(new_db_name, "%s/%s", root_dcfg->data_dir, var_name->value);
+            dbm_filename = apr_pstrcat(root_dcfg->mp, root_dcfg->data_dir, "/", var_name->value, NULL);
```

Comment:
why we need apr_pstrcat, for memory issue?

Reply:
Yes. apr_pstrcat will allocate memory for dbm_filename and we don't need to care the memory issue.